spec/unit/facter/plugins_spec.rb

```diff
@@ -11,7 +11,11 @@
     Facter.clear
     Facter.loadfacts
     allow(Puppet::Jenkins::Plugins).to receive(:available).and_return(plugins)
-    Facter.fact(:kernel).stub(:value).and_return(kernel)
+
+    overridden_kernel = kernel
+    Facter.add(:kernel, weight: 9999) do
```

Comment:
I am not quite sure this change is required. could you explain it a bit?

Reply:
I've reverted it back to stubbing.